ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_15/EventQueryServiceIT.scala

```diff
@@ -0,0 +1,269 @@
+
+package com.daml.ledger.api.testtool.suites.v1_15
+
+import com.daml.ledger.api.refinements.ApiTypes
+import com.daml.ledger.api.testtool.infrastructure.Allocation._
+import com.daml.ledger.api.testtool.infrastructure.Assertions._
+import com.daml.ledger.api.testtool.infrastructure.LedgerTestSuite
+import com.daml.ledger.api.v1.event_query_service.{
+  GetEventsByContractIdRequest,
+  GetEventsByContractKeyRequest,
+}
+import com.daml.ledger.api.v1.value._
+import com.daml.ledger.test.model.Test.{Dummy, _}
+import com.daml.lf.value.Value.ContractId
+import scalapb.GeneratedMessage
+import scalaz.Tag
+import scalaz.syntax.tag.ToTagOps
+
+import scala.collection.immutable
+import scala.concurrent.Future
+
+class EventQueryServiceIT extends LedgerTestSuite {
+
+  // Note that the Daml template must be inspected to establish the key type and fields
+  // For the TextKey template the key is: (tkParty, tkKey) : (Party, Text)
+  // When populating the Record identifiers are not required.
+  private def makeTextKeyKey(party: ApiTypes.Party, keyText: String) = {
+    Value(
+      Value.Sum.Record(
+        Record(fields =
+          Vector(
+            RecordField(value = Some(Value(Value.Sum.Party(party.unwrap)))),
+            RecordField(value = Some(Value(Value.Sum.Text(keyText)))),
+          )
+        )
+      )
+    )
+  }
+
+  test(
+    "TXEventsByContractIdBasic",
+    "Expose a create event by contract id",
+    allocate(SingleParty),
+  )(implicit ec => { case Participants(Participant(ledger, party)) =>
+    for {
+      tx <- ledger.submitAndWaitForTransaction(
+        ledger.submitAndWaitRequest(party, Dummy(party).create.command)
+      )
+      expected = assertDefined(
+        tx.transaction.flatMap(_.events.flatMap(_.event.created).headOption),
+        "Expected a created event",
+      )
+      events <- ledger.getEventsByContractId(
+        GetEventsByContractIdRequest(expected.contractId, Tag.unsubst(immutable.Seq(party)))
+      )
+    } yield {
+      val actual = assertDefined(events.createEvent, "Expected a created event")
+      assertEquals("Looked up event should match the transaction event", actual, expected)
+    }
+  })
+
+  test(
+    "TXEventsByContractIdConsumed",
+    "Expose an archive event by contract id",
+    allocate(SingleParty),
+  )(implicit ec => { case Participants(Participant(ledger, party)) =>
+    for {
+      dummyCid <- ledger.create(party, Dummy(party))
+      tx <- ledger.submitAndWaitForTransaction(
+        ledger.submitAndWaitRequest(party, dummyCid.exerciseDummyChoice1().command)
+      )
+      expected = assertDefined(
+        tx.getTransaction.events.flatMap(_.event.archived).headOption,
+        "Expected an exercised event",
+      )
+      events <- ledger.getEventsByContractId(
+        GetEventsByContractIdRequest(dummyCid.unwrap, immutable.Seq(party.unwrap))
+      )
+    } yield {
+      val actual = assertDefined(events.archiveEvent, "Expected a exercise event")
```

Comment:
I think we should check that we also populated the createdEvent in this case. We can't assume that the code will always stay correct in that regard.